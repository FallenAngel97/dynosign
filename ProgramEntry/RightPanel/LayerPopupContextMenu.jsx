import React from 'react';
import phoneLayout from '../MainArea/phone_layout.svg';
import desktopLayout from '../MainArea/desktop_layout.svg';
import './LayerPopupContextMenu.scss';
import PropTypes from 'prop-types';

export default class LayerPopupContextMenu extends React.Component {
  constructor (props) {
    super(props);
    this.dismissPopup = this.dismissPopup.bind(this);
  }
  componentDidMount () {
    document.addEventListener('click', this.dismissPopup, false)
  }
  componentWillUnmount () {
    document.removeEventListener('click', this.dismissPopup, false)
  }
  dismissPopup (e) {
    if (!this.node.contains(e.target)) {
      this.props.dismissPopup();
    }
  }
  render () {
    return (
      <div ref={ node => { this.node = node } } style={{ display: !this.props.hidden ? 'block' : 'none' }} id='layer_popup_contextmenu'>
        <div id='layout_context_selector'>
          <img src={phoneLayout} />
          <img src={desktopLayout} />
        </div>
        <ul>
          <li>Move up</li>
          <li>Move down</li>
          <li>Style</li>
        </ul>
      </div>
    )
  }
}
LayerPopupContextMenu.propTypes = {
  hidden: PropTypes.bool,
  dismissPopup: PropTypes.func
}
