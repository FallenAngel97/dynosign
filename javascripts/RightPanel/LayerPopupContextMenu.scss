#layer_popup_contextmenu {
    position: absolute;
    width: 100px;
    background: #e4e4e4;
    box-shadow: 2px 2px 11px 1px rgba(0, 0, 0, 0.2);
    padding: 3px;
    ul {
        list-style: none;
        padding-left: 0;
    }
}
#layout_context_selector {
    display: flex;
    justify-content: space-around;
    border-bottom: 1px solid #bfbfbf;
    padding-bottom: 15px;
    padding-top: 14px;
    img {
        height: 20px;
    }
}