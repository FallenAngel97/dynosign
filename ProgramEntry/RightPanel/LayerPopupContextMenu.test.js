/* global shallow */
import React from 'react';
import LayerPopupContextMenu from './LayerPopupContextMenu';

describe('LayerPopupContextMenu tests', () => {
  test('LayerPopupContextMenu properly renders', () => {
    let layerPopupContextMenu = shallow(<LayerPopupContextMenu />);
    expect(layerPopupContextMenu).toMatchSnapshot();
  });
});
