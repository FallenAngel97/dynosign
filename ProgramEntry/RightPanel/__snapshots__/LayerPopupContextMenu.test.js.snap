// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayerPopupContextMenu tests LayerPopupContextMenu properly renders 1`] = `ShallowWrapper {}`;
